### ISSUE TITLE
[6.0.0-rc.4] arraySplice doesn't work properly

### ISSUE THREAD

**Issue by clayne11** [erikras/redux-form#1490]
Calling `arraySplice` with the plain structure doesn't work properly. If you pass a `removeNum` and a `value`, then the values get removed but nothing gets added. This is the logic:

```javascript
const splice = (array = [], index, removeNum, value) => {
  const copy = [ ...array ]
  if (removeNum) {
    copy.splice(index, removeNum)   // removing
  } else {
    if(index < copy.length) {
      copy.splice(index, 0, value)  // adding
    } else {
      copy[index] = value           // outside range, so just set it
    }
  }
  return copy
}

export default splice
```

In the first branch it should be `copy.splice(index, removeNum, value)`.

The logic for `ImmutableJS` isn't correct either:

```javascript
  splice: (list = List.isList(list) || List(), index, removeNum, value) => {
    if(removeNum) {
      return list.splice(index, removeNum)  // removing
    }
    if(index < list.size) {
      return list.splice(index, 0, value)   // inserting
    }
    return list.set(index, value)         // inserting after end of list
  }
```

When adding using this method it will first remove a value then override the remaining value in that position.

**Comment by clayne11** [comment]
Could you possibly cut a release on this today? I need to deploy a new feature and I'm held up with this.

I also posted 2 more small PRs yesterday. It would be great if all of them could go into a new release.

**Comment by erikras** [comment]
Fix published in [`v6.0.0-rc.5`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-rc.5).
